cmd/manager/main.go

```diff
@@ -92,6 +94,18 @@ func main() {
 	log.Print("Starting the Cmd.")
 	stopCh := signals.SetupSignalHandler()
 
+	// set clusterOperator status to report Migration
+	status.ReportMigration(cfg, mgr, namespace, os.Getenv("RELEASE_VERSION"), stopCh)
+	// Perform migration logic to upgrade cluster from 4.1.x to 4.2.0
+	client, err := client.New(config.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		exit(err)
```

Comment:
Do we really want to terminate the operator if we cannot migrate? Even if the CVO restarts the operator, we could end up in the same situation again. This will basically result in a cluster that doesn't have a marketplace operator. The same comment applies for the `Migrate()` call.

Reply:
It seems like the right thing to do would be to retry until it succeeds, given the normal pattern of eventual consistency.

Reply:
What if it is stuck here for whatever reason? We will be migrating for ever. So either we do this in a background goroutine or do a best effort migration.

Reply:
> What if it is stuck here for whatever reason? 

Exactly my concern. Do we want to keep consuming resources in an effort to migrate, if we are unable to migrate fully?

Reply:
Also, the Subscriptions are updated by the migrator, so unless there's a switch which indicates if a Subscription has been updated or not, all the Subscriptions will keep getting updated, albeit just one field.

Keeping track of which Subscriptions were updated means consuming even more resources.

Reply:
Then we should be removing the labels on the Subscriptions that we update

Reply:
I think that doing it every time we start is just part of the deal. I'm not sure how else we would handle it.

Are we okay with our migration failing and then moving on? Perhaps it depends on our definition of `failure`?

Reply:
> Then we should be removing the labels on the Subscriptions that we update

@aravindhp Are we sure we can do that without producing any side effects? 

> Are we okay with our migration failing and then moving on? Perhaps it depends on our definition of failure?

Maybe clean up/ update as many resources possible, and report resources that could not be cleaned/updated for admin to act on them?

Reply:
>Are we sure we can do that without producing any side effects?

Yes. Note that this does not mean removing *all* labels. Just the ones we care about.

>Maybe clean up/ update as many resources possible,

This is what we should be doing in general.

> report resources that could not be cleaned/updated for admin to act on them?

How will you report? Asking admins to look at logs is not acceptable. Please think of worst case scenarios where not updating a Subscription or not being able to delete the datastore/installed `CatalogSourceConfigs` will cause issues.

Reply:
> Please think of worst case scenarios where not updating a Subscription or not being able to delete the datastore/installed CatalogSourceConfigs will cause issues.

Does that mean we're agreeing on creating a go routine that keeps making best effort to clean up resources until there's no errors?

Reply:
You shouldn't jump  to that conclusion until you explore the scenarios.